sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs

```diff
@@ -42,668 +33,327 @@ public MetricsAdvisorClient(Uri endpoint, MetricsAdvisorKeyCredential credential
+        /// Adds a <see cref="MetricFeedback"/>.
         /// </summary>
-        /// <param name="metricId">The unique identifier of the <see cref="DataFeedMetric"/>.</param>
-        /// <param name="options">The set of options used to configure the request's behavior.</param>
+        /// <param name="feedback">The <see cref="MetricFeedback"/> to be created.</param>
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
-        /// <returns>An <see cref="AsyncPageable{T}"/> containing the collection of <see cref="MetricSeriesDefinition"/>s.</returns>
-        /// <exception cref="ArgumentNullException"><paramref name="metricId"/> or <paramref name="options"/> is null.</exception>
-        /// <exception cref="ArgumentException"><paramref name="metricId"/> is empty or not a valid GUID.</exception>
-        public virtual AsyncPageable<MetricSeriesDefinition> GetMetricSeriesDefinitionsAsync(string metricId, GetMetricSeriesDefinitionsOptions options, CancellationToken cancellationToken = default)
+        /// <returns>
+        /// A <see cref="Response{T}"/> containing the result of the operation. The result is a <see cref="MetricFeedback"/>
+        /// containing information about the newly added feedback.
+        /// </returns>
+        /// <exception cref="ArgumentNullException"><paramref name="feedback"/> is null.</exception>
+        public virtual async Task<Response<MetricFeedback>> AddFeedbackAsync(MetricFeedback feedback, CancellationToken cancellationToken = default)
         {
-            Argument.AssertNotNull(options, nameof(options));
-
-            Guid metricGuid = ClientCommon.ValidateGuid(metricId, nameof(metricId));
-            MetricSeriesQueryOptions queryOptions = new MetricSeriesQueryOptions(ClientCommon.NormalizeDateTimeOffset(options.ActiveSince));
-
-            int? skip = options.Skip;
-            int? maxPageSize = options.MaxPageSize;
+            Argument.AssertNotNull(feedback, nameof(feedback));
 
-            // Deep copy filter contents from options to queryOptions.
+            using DiagnosticScope scope = ClientDiagnostics.CreateScope($"{nameof(MetricsAdvisorClient)}.{nameof(AddFeedback)}");
+            scope.Start();
 
-            foreach (KeyValuePair<string, IList<string>> kvp in options.DimensionCombinationsFilter)
+            try
             {
-                queryOptions.DimensionFilter.Add(kvp.Key, new List<string>(kvp.Value));
-            }
+                RequestContent content = MetricFeedback.ToRequestContent(feedback);
+                RequestContext context = new RequestContext()
+                {
+                    CancellationToken = cancellationToken,
+                };
+                Response response = await CreateMetricFeedbackAsync(content, context).ConfigureAwait(false);
+                var headers = new MicrosoftAzureMetricsAdvisorRestAPIOpenAPIV2CreateMetricFeedbackHeaders(response);
```

Comment:
If we were in a DPG-only world, we wouldn't have the header models, would we?  If not, we should try to write this method without it as a proof of concept.

Reply:
Removed headers!